### PR TITLE
✨💄(Student Profile): Added UI Responsive, Linked API to UI

### PR DESCRIPTION
**Pull request by caaaramia**
Added mobile responsiveness to all components within the student-profile feature and created and linked API calls from the API backend to the frontend to populate the page with mock data.

### CODE REVIEW

File: libs/client/student-profile/feature/src/lib/profile-card/profile-card.component.html [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
...7.1-39 1.7-4.3 7.4-22-1.7-45-13.9-4.3-45.7 17.9-45.7 17.9-13.2-3.7-27.5-5.6-41.6-5.6-14.1 0-28.4 1.9-41.6 5.6 0 0-31.8-22.2-45.7-17.9-9.1 22.9-3.5 40.6-1.7 45-10.6 11.7-15.6 20.8-15.6 39 0 63.6 37.3 69 74.3 73.1-4.8 4.3-9.1 11.7-10.6 22.3-9.5 4.3-33.8 11.7-48.3-13.9-9.1-15.8-25.5-17.1-25.5-17.1-16.2-.2-1.1 10.2-1.1 10.2 10.8 5 18.4 24.2 18.4 24.2 9.7 29.7 56.1 19.7 56.1 19.7 0 13.9.2 36.5.2 40.6 0 4.3-3 9.5-11.5 8-66-22.1-112.2-84.9-112.2-158.3 0-91.8 70.2-161.5 162-161.5S388 165.6 388 257.4c.1 73.4-44.7 136.3-110.7 158.3zm-98.1-61.1c-1.9.4-3.7-.4-3.9-1.7-.2-1.5 1.1-2.8 3-3.2 1.9-.2 3.7.6 3.9 1.9.3 1.3-1 2.6-3 3zm-9.5-.9c0 1.3-1.5 2.4-3.5 2.4-2.2.2-3.7-.9-3.7-2.4 0-1.3 1.5-2.4 3.5-2.4 1.9-.2 3.7.9 3.7 2.4zm-13.7-1.1c-.4 1.3-2.4 1.9-4.1 1.3-1.9-.4-3.2-1.9-2.8-3.2.4-1.3 2.4-1.9 4.1-1.5 2 .6 3.3 2.1 2.8 3.4zm-12.3-5.4c-.9 1.1-2.8.9-4.3-.6-1.5-1.3-1.9-3.2-.9-4.1.9-1.1 2.8-.9 4.3.6 1.3 1.3 1.8 3.3.9 4.1zm-9.1-9.1c-.9.6-2.6 0-3.7-1.5s-1.1-3.2 0-3.9c1.1-.9 2.8-.2 3.7 1.3 1.1 1.5 1.1 3.3 0 4.1zm-6.5-9.7c-.9.9-2.4.4-3.5-.6-1.1-1.3-1.3-2.8-.4-3.5.9-.9 2.4-.4 3.5.6 1.1 1.3 1.3 2.8.4 3.5zm-6.7-7.4c-.4.9-1.7 1.1-2.8.4-1.3-.6-1.9-1.7-1.5-2.6.4-.6 1.5-.9 2.8-.4 1.3.7 1.9 1.8 1.5 2.6z"/></svg></a>
```
Comment (DanieldeNobrega) [inline libs/client/student-profile/feature/src/lib/profile-card/profile-card.component.html:27]: Can you please explain what is happening in this file?

File: libs/client/student-profile/feature/src/lib/location/location.component.ts [hunk 2]
```diff
@@ -1,16 +1,83 @@
...
 })
 export class LocationComponent implements OnInit {
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  constructor() { }
+  
+  constructor() { 
+    this.location = null;
+  }
+
+  @Input() location: any;
```
Comment (DanieldeNobrega) [inline libs/client/student-profile/feature/src/lib/location/location.component.ts:15]: There is a warning that states, "Unexpected any. Specify a different type", whenever you try and use the word "any".
Will this warning effect code in the future?

Comment (caaaramia) [reply libs/client/student-profile/feature/src/lib/location/location.component.ts:15]: The warning won't impact anything, but will look into a way to get rid of it.

File: libs/client/student-profile/feature/src/lib/profile-card/profile-card.component.html (hunk 1 above)
Comment (caaaramia) [reply libs/client/student-profile/feature/src/lib/profile-card/profile-card.component.html:27]: The <a> tags are for the social media links. Code was taken from the Company-Profile feature to allow for consistency in design between the profiles. Basically just icons for the links.

**Review by DanieldeNobrega — APPROVED** [review]
All files look good, you can go ahead and merge.

**Pull request merged**